### ISSUE TITLE
aws_dynamodb_table: local secondary index definition not sending up NonKeyAttributes

### ISSUE THREAD

**Issue by ballwood** [terraform-providers/terraform-provider-aws#15115]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform v0.12.19
+ provider.aws v3.6.0

### Affected Resource(s)

* aws_dynamodb_table

### Terraform Configuration Files

```hcl
resource "aws_dynamodb_table" "agent_jobs" {
  name = "data-agents2"
  hash_key  = "agentId"
  range_key = "jobId"

  attribute {
    name = "jobId" // job id
    type = "S"
  }

  attribute {
    name = "agentId"
    type = "S"
  }

  attribute {
    name = "jobStatusDate"
    type = "S"
  }

  local_secondary_index {
    name               = "idx-agentId-jobStatusDate"
    projection_type    = "INCLUDE"
    range_key          = "jobStatusDate"
    non_key_attributes = ["revision"]
  }

  billing_mode     = "PAY_PER_REQUEST"
  stream_enabled   = true
  stream_view_type = "NEW_AND_OLD_IMAGES"
}
```

### Debug Output

Request sent to aws:
````
{
  "AttributeDefinitions": [
    {
      "AttributeName": "jobStatusDate",
      "AttributeType": "S"
    },
    {
      "AttributeName": "agentId",
      "AttributeType": "S"
    },
    {
      "AttributeName": "jobId",
      "AttributeType": "S"
    }
  ],
  "BillingMode": "PAY_PER_REQUEST",
  "KeySchema": [
    {
      "AttributeName": "agentId",
      "KeyType": "HASH"
    },
    {
      "AttributeName": "jobId",
      "KeyType": "RANGE"
    }
  ],
  "LocalSecondaryIndexes": [
    {
      "IndexName": "idx-agentId-jobStatusDate",
      "KeySchema": [
        {
          "AttributeName": "agentId",
          "KeyType": "HASH"
        },
        {
          "AttributeName": "jobStatusDate",
          "KeyType": "RANGE"
        }
      ],
      "Projection": {
        "ProjectionType": "INCLUDE"
      }
    }
  ],
  "StreamSpecification": {
    "StreamEnabled": true,
    "StreamViewType": "NEW_AND_OLD_IMAGES"
  },
  "TableName": "data-agents2",
  "Tags": []
}
````

### Expected Behavior

Should have created the dynamodb table

### Actual Behavior

Did not create the table. Error: ProjectionType is INCLUDE, but NonKeyAttributes is not specified. Non-key attributes were specified in HCL

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by ballwood** [comment]
Investigated and was broken due to https://github.com/terraform-providers/terraform-provider-aws/issues/15115 as reverting to 3.4.0 works fine

**Comment by anGie44** [comment]
Hi @ballwood, thank you for creating this issue and apologies you came across this regression! I believe it's related to changes made in https://github.com/terraform-providers/terraform-provider-aws/pull/9988 , such that I removed the logic to expand that list of `non_key_attributes`, explaining why the request no longer contains that field anymore.

Relevant code that needs to be added back:
https://github.com/terraform-providers/terraform-provider-aws/pull/9988/files#diff-609906472a0a9de4d53f3baeae057583L4211-L4212

**Comment by anGie44** [comment]
Hi @ballwood , the fix for this has been merged and will release with `v3.8.0` of the Terraform AWS Provider, expected out next Thursday 👍

**Comment by ballwood** [comment]
Thanks Angie 😊

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by omkar-k3** [comment]
I am also facing same issue. I tried to create LSI from console and the same logic has been put into my terraform code but it throws an error 'Projection type is INCUDE but NonKeyAttributes is not specified'
Is fix coming in next couple of days?.

**Comment by anGie44** [comment]
Hi @omkar-k3 , thanks for commenting here. Yes, the fix has been merged and will likely be released this Thursday (`v3.8.0` of the AWS Provider).
